<h2>Evaluación Diagnostica Git/Gitflow IIC2154</h2>
<p>Alejandro Rico Rozo - Sección 4</p>

In [1]:
from functions import (
    most_retweeted,
    most_active_users,
    most_active_days,
    most_frecuent_hashtags
)

In [2]:
from process_file import users, tweets, get_raw_tweets

raw_tweets = get_raw_tweets()
users = users(raw_tweets)
tweets = tweets(raw_tweets)

In [3]:
users.head(2)

username        displayname               userId  \
0   ShashiRajbhar6     Shashi Rajbhar  1015969769760096256   
1  kaursuk06272818  KAUR SUKH🌾ਕੌਰ ਸੁਖ  1332937272581263362   

                                         description  \
0  Satya presan 🤔ho Sakta but prajit💪 nhi\njhuth ...   
1        ਜਿਓਣਾ ਕੀ ਸਰੀਰਾਂ ਦਾ ਜੇਕਰ ਹੋਣ ਜ਼ਮੀਰਾਂ ਮਰੀਆਂ 🌼   

                                      rawDescription descriptionUrls  \
0  Satya presan 🤔ho Sakta but prajit💪 nhi\njhuth ...              []   
1        ਜਿਓਣਾ ਕੀ ਸਰੀਰਾਂ ਦਾ ਜੇਕਰ ਹੋਣ ਜ਼ਮੀਰਾਂ ਮਰੀਆਂ 🌼              []   

   verified                    created  followersCount  friendsCount  ...  \
0     False  2018-07-08T14:44:03+00:00            1788          1576  ...   
1     False  2020-11-29T06:40:06+00:00              51            68  ...   

   favouritesCount  listedCount  mediaCount                  location  \
0            26071            1         254  Azm Uttar Pradesh, India   
1             3676            0         607                             

  protected  linkUrl linkTcourl  \
0     False     None       None   
1     False     None       None   

                                     profileImageUrl  \
0  https://pbs.twimg.com/profile_images/135433129...   
1  https://pbs.twimg.com/profile_images/133295149...   

                                    profileBannerUrl  \
0  https://pbs.twimg.com/profile_banners/10159697...   
1  https://pbs.twimg.com/profile_banners/13329372...   

                            profileUrl  
0   https://twitter.com/ShashiRajbhar6  
1  https://twitter.com/kaursuk06272818  

[2 rows x 21 columns]

In [4]:
tweets.head(2)

tweetUrl  \
0  https://twitter.com/ShashiRajbhar6/status/1376...   
1  https://twitter.com/kaursuk06272818/status/137...   

                       date  \
0 2021-03-30 03:33:46+00:00   
1 2021-03-30 03:33:23+00:00   

                                             content  \
0                       Support 👇\n\n#FarmersProtest   
1  Supporting farmers means supporting our countr...   

                                     renderedContent              tweetId  \
0                       Support 👇\n\n#FarmersProtest  1376739399593910273   
1  Supporting farmers means supporting our countr...  1376739306287427584   

                                                user outlinks tcooutlinks  \
0  {'username': 'ShashiRajbhar6', 'displayname': ...       []          []   
1  {'username': 'kaursuk06272818', 'displayname':...       []          []   

   replyCount  retweetCount  ...       conversationId  lang  \
0           0             0  ...  1376739399593910273    en   
1           0             0  ...  1376739306287427584    en   

                                              source  \
0  <a href="http://twitter.com/download/android" ...   
1  <a href="http://twitter.com/download/android" ...   

                             sourceUrl          sourceLabel  \
0  http://twitter.com/download/android  Twitter for Android   
1  http://twitter.com/download/android  Twitter for Android   

                                               media retweetedTweet  \
0                                               None            NaN   
1  [{'previewUrl': 'https://pbs.twimg.com/media/E...            NaN   

  quotedTweet  mentionedUsers               userId  
0        None            None  1015969769760096256  
1        None            None  1332937272581263362  

[2 rows x 22 columns]

In [5]:
top_10_tweets = most_retweeted(tweets)
top_10_tweets

content  retweetCount
408128  why aren’t we talking about this?! #FarmersPro...        315547
395142  We stand in solidarity with the #FarmersProtes...        103957
266196  I still #StandWithFarmers and support their pe...         67694
366579  “Paid actors,” huh? Quite the casting director...         35921
372793  What in the human rights violations is going o...         26972
314192  Happy to share that I’ve donated $10,000 to pr...         23251
215034  There has been much social media coverage arou...         20132
398011  Farmers feed the world. Fight for them. Protec...         18744
325261  Should be talking about this! #FarmersProtest\...         17368
163689  To all of my influencer/celeb friends- read up...         15677

In [6]:
top_10_users = most_active_users(tweets, users)
top_10_users

username                                        displayname  \
8      DigitalKisanBot                                        Kisan Bot🚜🌾   
15            Jass_k_G                                              JassG   
32       rebelpacifist                                Theinactiveactivist   
75       tasveersandhu  ਕਿਸਾਨAndolajivi ravinder kaur चोकीदार ही कातिल है   
141             jot__b                                                Jot   
146        Iamjazzie96                                 Jaz 🇨🇦🌾ਗਰਮ ਖਿਆਲੀ 📌   
317           rumsomal                                            🍊raman🚜   
1394   shells_n_petals  Jaspal Kaur Bains.For love of Punjab,Sikhi & I...   
41753    z_khalique007                                  mohd khaliquzzama   
76038  harjot_tweeting                                       Harjot Singh   

                    userId  
8      1309302936317636608  
15     1340186972615307264  
32     1354532795847073796  
75                69027875  
141              452391771  
146     872085660886282240  
317               48085084  
1394            2930682630  
41753   921732117024919552  
76038   980633168981061632

In [7]:
top_10_days = most_active_days(tweets)
top_10_days

2021-02-03    83866
2021-02-04    58607
2021-02-05    33332
2021-02-02    28548
2021-02-06    22420
2021-02-07    11325
2021-02-09     9320
2021-02-08     8920
2021-02-10     7973
2021-02-11     5698
Name: date_day, dtype: int64

<p>2021-02-03 : 83866</p>
<p>2021-02-04 : 58607</p>
<p>2021-02-05 : 33332</p>
<p>2021-02-02 : 28548</p>
<p>2021-02-06 : 22420</p>
<p>2021-02-07 : 11325</p>
<p>2021-02-09 :  9320</p>
<p>2021-02-08 :  8920</p>
<p>2021-02-10 :  7973</p>
<p>2021-02-11 :  5698</p>
<p>Name: date_day, dtype: int64</p>

In [8]:
top_10_hashtags = most_frecuent_hashtags(tweets)
top_10_hashtags

{'FarmersProtest': 392372,
 'IStandWithFarmers': 15618,
 'farmersprotest': 14393,
 'IndianFarmersHumanRights': 11897,
 'FarmersAreIndia': 10952,
 'StandWithFarmers': 10446,
 'Rihanna': 8796,
 'FarmersProtests': 8646,
 'Farmers': 6255,
 'shameonbollywood': 6187}

<p>{</p>
<p> 'FarmersProtest': 392372,</p>
<p> 'IStandWithFarmers': 15618,</p>
<p> 'farmersprotest': 14393,</p>
<p> 'IndianFarmersHumanRights': 11897,</p>
<p> 'FarmersAreIndia': 10952,</p>
<p> 'StandWithFarmers': 10446,</p>
<p> 'Rihanna': 8796,</p>
<p> 'FarmersProtests': 8646,</p>
<p> 'Farmers': 6255,</p>
<p> 'shameonbollywood': 6187</p>
<p>}</p>